modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AuthenticationClient.java

```diff
@@ -46,24 +50,26 @@ public String getEndpoint() {
     }
 
     public String getOAuth2AccessToken(String username, String password, String clientId, String clientPassword) {
-        String body = String.format("%s=%s&%s=%s&%s=%s", SW360Attributes.AUTHENTICATOR_GRANT_TYPE, GRANT_TYPE_VALUE,
-                SW360Attributes.AUTHENTICATOR_USERNAME, username, SW360Attributes.AUTHENTICATOR_PASSWORD, password);
+        try {
+            String body = String.format("%s=%s&%s=%s&%s=%s", SW360Attributes.AUTHENTICATOR_GRANT_TYPE, GRANT_TYPE_VALUE,
+                    SW360Attributes.AUTHENTICATOR_USERNAME, username, SW360Attributes.AUTHENTICATOR_PASSWORD, password);
 
-        HttpHeaders headers = new HttpHeaders();
-        addBasicAuthentication(headers, clientId, clientPassword);
-        headers.add(HttpHeaders.CONTENT_TYPE, TOKEN_CONTENT_TYPE);
-
-        HttpEntity<String> httpEntity = new HttpEntity<>(body, headers);
-        ResponseEntity<String> response = doRestCall(getEndpoint(), HttpMethod.POST, httpEntity, String.class);
-
-        if (response.getStatusCode() == HttpStatus.OK) {
-            try {
-                return (String) new ObjectMapper().readValue(response.getBody(), HashMap.class).get(JSON_TOKEN_KEY);
-            } catch (IOException e) {
-                throw new ExecutionException("Error when attempting to deserialize the response body.", e);
-            }
-        } else {
-            throw new ExecutionException("Could not request OAuth2 access token for [" + username + "].");
+            HttpHeaders headers = new HttpHeaders();
+            addBasicAuthentication(headers, clientId, clientPassword);
+            headers.add(HttpHeaders.CONTENT_TYPE, TOKEN_CONTENT_TYPE);
+
+            HttpEntity<String> httpEntity = new HttpEntity<>(body, headers);
+
+            ResponseEntity<String> response = doRestCall(getEndpoint(), HttpMethod.POST, httpEntity, String.class);
+
+            checkRestStatus(response);
+            return (String) new ObjectMapper().readValue(response.getBody(), HashMap.class).get(JSON_TOKEN_KEY);
```

Comment:
What is returned here if the body is null?

Reply:
readValue() has an internal null check for its content.